code-challenges.test.js

```diff
@@ -28,30 +37,119 @@ const colors2 = [
+}
 // Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
+const votes2 = { 
+  upVotes: 2, 
+  downVotes: 33 
+}
 // Expected output: -31
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+//Function Name: totalVotes
+//Input: Objects
+//Output: Number
+//Process: Create a function called totalVotes. Use the argument object.valueName to get just the number from the object. Then subtract it to get the final answer. 
+
+const totalVotes = (object) => {
+  return object.upVotes - object.downVotes
+}
+
+// console.log(totalVotes(votes1))
+// console.log(totalVotes(votes2))
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
+describe("fixArray", () => {
+  it("Returns an array that should repeat words from previous two arrays, but only have the same words once on the new array.", () => {
+    expect(fixArray(...dataTypesArray1,...dataTypesArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
 
 const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
 const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
+
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+//Function Name: fixArray
+//Input: Two arrays
+//Output: One array
+//Process: Create a function called fixArray. Take in two arrays in the argument. Create a variable and within it use filter to filter through an a array and have it check what values we don't want duplicated. Return the value of everything that does not include "string" and "Boolean". Outside of the .filter return both arrays combined. 
+
+//First Try
+// Could not get it to run with JEST! Could you explain, please. 
+// const fixArray = (arr1, arr2) => {
+//    const filterArr = arr2.filter((value) => {
+//     return ! value.includes("string") && ! value.includes("Boolean")
+//   })
+//   return arr1.concat(filterArr)
+// }
+//console.log(fixArray(...dataTypesArray1,...dataTypesArray2))
+
+// Pseudo code:
+//Function Name: fixArray
+//Input: Two arrays
+//Output: One array
+//Process: Create a function called fixArray. Use ...to take in iteration of an array or string. Filter through the array and check the values as well as the index. When iterating through the array we will return an array which will return the values of the array, but if there is a duplication then the value will not be returned. 
+
+//Second Try
+//Had to get help, but would my first try work?
+const fixArray = (...arr) => {
+  return arr.filter((value, index) => {
+    return arr.indexOf(value) == index //If the value appears more than once then .indexOf will take the first word of it. 
```

Comment:
This function works for the current test, with the spread operators in the argument, but will not work if the test is set up properly as I wrote above.